test/Quoter.t.sol

```diff
@@ -0,0 +1,754 @@
+
+        assertEq(deltaAmounts[0], 15273);
+        assertEq(sqrtPriceX96AfterList[0], 80418414376567919517220409857);
+        assertEq(initializedTicksLoadedList.length, 1);
+        assertEq(initializedTicksLoadedList[0], 2);
+    }
+
+    function testQuoter_quoteExactOutput_2to0_2TicksLoaded_initialiedAfter() public {
+        tokenPath.push(token2);
+        tokenPath.push(token0);
+
+        ExactOutputParams memory params = getExactOutputParams(tokenPath, 6223);
+
+        (
+            int128[] memory deltaAmounts,
+            uint160[] memory sqrtPriceX96AfterList,
+            uint32[] memory initializedTicksLoadedList
+        ) = quoter.quoteExactOutput(params);
+
+        assertEq(deltaAmounts[0], 6283);
+        assertEq(sqrtPriceX96AfterList[0], 79708304437530892332449657932);
+        assertEq(initializedTicksLoadedList.length, 1);
+        assertEq(initializedTicksLoadedList[0], 2);
+    }
+
+    function testQuoter_quoteExactOutput_2to0_1TickLoaded() public {
+        tokenPath.push(token2);
+        tokenPath.push(token0);
+
+        ExactOutputParams memory params = getExactOutputParams(tokenPath, 6000);
+        (
+            int128[] memory deltaAmounts,
+            uint160[] memory sqrtPriceX96AfterList,
+            uint32[] memory initializedTicksLoadedList
+        ) = quoter.quoteExactOutput(params);
+
+        assertEq(deltaAmounts[0], 6055);
+        assertEq(sqrtPriceX96AfterList[0], 79690640184021170956740081887);
+        assertEq(initializedTicksLoadedList.length, 1);
+        assertEq(initializedTicksLoadedList[0], 1);
+    }
+
+    function testQuoter_quoteExactOutput_2to1() public {
+        tokenPath.push(token2);
+        tokenPath.push(token1);
+
+        ExactOutputParams memory params = getExactOutputParams(tokenPath, 9871);
+
+        (
+            int128[] memory deltaAmounts,
+            uint160[] memory sqrtPriceX96AfterList,
+            uint32[] memory initializedTicksLoadedList
+        ) = quoter.quoteExactOutput(params);
+
+        assertEq(deltaAmounts[0], 10000);
+        assertEq(sqrtPriceX96AfterList[0], 80018020393569259756601362385);
+        assertEq(initializedTicksLoadedList.length, 1);
+        assertEq(initializedTicksLoadedList[0], 0);
+    }
+
+    function testQuoter_quoteExactOutput_0to2to1() public {
+        tokenPath.push(token0);
+        tokenPath.push(token2);
+        tokenPath.push(token1);
+
+        ExactOutputParams memory params = getExactOutputParams(tokenPath, 9745);
+
+        (
+            int128[] memory deltaAmounts,
+            uint160[] memory sqrtPriceX96AfterList,
+            uint32[] memory initializedTicksLoadedList
+        ) = quoter.quoteExactOutput(params);
+
+        assertEq(deltaAmounts[0], 10000);
+        assertEq(deltaAmounts[1], 0);
+        assertEq(deltaAmounts[2], -9745);
+        assertEq(sqrtPriceX96AfterList[0], 78461888503179331029803316753);
+        assertEq(sqrtPriceX96AfterList[1], 80007838904387594703933785072);
```

Comment:
lil sus that this reverse order of the equivalent v3 quoter test: https://github.com/Uniswap/v3-periphery/blob/697c2474757ea89fec12a4e6db16a574fe259610/test/QuoterV2.spec.ts#L528C9-L528C80

Reply:
now I think about it, this behavior is fine. Take `deltaAmounts` for example, trade path is 0 -> 2 -> 1 so `deltaAmounts[0]` is positive (into the pool) while `deltaAmounts[2]` is negative (taken out of the pool)

Reply:
just to clarify, there is no right or wrong here just different behavior and I the ordering I have more